test/functionality/typing/constant-type-inference.test.ts

```diff
@@ -0,0 +1,33 @@
+import { beforeAll, describe, expect, test } from 'vitest';
+import { retrieveNormalizedAst, withShell } from '../_helper/shell';
+import type { NormalizedAst } from '../../../src/r-bridge/lang-4.x/ast/model/processing/decorate';
+import { RDataType, TypeInferencer } from '../../../src/typing/type-inferencer';
+
+describe.sequential('Infer types for constant expressions', withShell(shell => {
```

Comment:
I think it is safe to switch to the Tree-Sitter engine for the tests (which should speed them up as well :D). We can discuss this in the next meeting if you want